scripts/docker-cross-tag.sh

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env bash
+
+# SPDX-License-Identifier: GPL-2.0-or-later
+# Copyright (C) 2024-2024  The DOSBox Staging Team
+
+usage() {
+    printf "%s\n" "\
+Usage: $0 FUNC OPTIONS
+
+Utility script to assist with docker image tags
+
+FUNC must be one of:
+  tag-exists         Test if image tag exists in GHCR.
+                     Takes two options: IMAGE_NAME, TAG.
+
+  hash-files         Compute a hash of specified files for use in a docker image tag
+                     docker image tag. Command takes one or more file paths as options.
+"
+}
+
+tag_exists_in_ghcr() {
+    local image="$1"
+    local tag="$2"
+
+    # jq and curl are required
+    if ! command -v jq &> /dev/null ; then
+        printf "%s\n" "jq not found!" && exit 1
+    fi
+    if ! command -v curl &> /dev/null ; then
+        printf "%s\n" "curl not found!" && exit 1
+    fi
+
+    # Get an anonymous token from Github to make further API calls.
+    # No need for accounts as the images are public.
+    token=$(curl -s https://ghcr.io/token\?scope="repository:dosbox-staging/${image}:pull" | jq -r .token)
+    # Get the list of tags from the registry, and attempts to find provided tag
+    # in the list using jq. Prints 'true' if tag found, or 'false' otherwise
+    curl -s -H "Authorization: Bearer $token" "https://ghcr.io/v2/dosbox-staging/${image}/tags/list" \
+        | jq --arg T "$tag" '.tags | any(.==$T)'
+}
+
+files_hash() {
+    # Hash the hash of the contents of the provided list of files.
```

Comment:
I was surprised that you're hashing twice, so googled it and found this:

https://crypto.stackexchange.com/questions/50017/why-hashing-twice

So this is some kind of a mitigation against "length extension attacks", whatever that means, to improve security.

But we're not doing anything security related here, we just need an unique hash, so maybe use a single hash to remove confusion?

Reply:
Nah, nothing security related. It just seemed the easiest way to get a single hash of multiple files. When you run sha256sum on multiple files, it spits out a hash per file. The other way would be to concatenate the files before hashing. This seemed simpler to me but I dunno.

Reply:
Ok, I've decided to just `cat` the files and pass it to `sha256sum`. Will make the intention clearer I think.

Reply:
Also lost a `cut` that wasn't needed.